src/mode/graphics.rs

```diff
@@ -65,17 +63,15 @@ where
     /// Write out data to display
     pub fn flush(&mut self) -> Result<(), ()> {
         let display_size = self.properties.get_size();
-        let iface = self.properties.borrow_iface_mut();
 
+        // TODO: Why is this here? Once during initialisation should suffice?
```

Comment:
Correct, I'm pretty sure the chip will remember these settings until told otherwise. Move them to init and see what happens? Presumably charmode will want to set this dynamically at some point to support text positioning?

Reply:
Surely charmode will use the very same on the fly when cursor repositioning is requested. It's a readily available method so I don't see a problem with that. Only the static call on every flush is a bit weird, if we always draw a full frame the display should always end up at the origin. Maybe we should throw in some error handling to reset the position in case a frame could not be fully transmitted but I'm not sure what will happen then anyways...

Reply:
That's what I was getting at; I think it should live in `init()` for graphics mode, but char mode will probably make copious use of it elsewhere :). If some code changes the position somewhere, then the user has the responsibility to change it back for full screen rendering to work properly IMO. I'm envisaging scenarios where higher framerates are more important than correctness for graphics mode, so the less we do in `flush()` the better.

Reply:
? This change factors that into the `set_draw_area()` method which can easily be used from `init()` or on the fly by charmode if reposititioning of the cursor is requested. What do you see missing here?

Reply:
I think we're misunderstanding each other :). I'm agreeing with your code comment; calling `set_draw_area()` here in `flush()` is just a bit of safety and can probably be moved to `init()`. That said it might be worth leaving in `flush()` as, like you said, it ensures the framebuffer is drawn in the right position. It's safety vs speed. I'd rather have `set_draw_area()` called once in `init()` for speed reasons, but what do you think?

Reply:
My 2 points (worth a cent each):
- There're two possibilities why the local frame buffer position might end up not in line with the display buffer: A (temporary) communication glitch or someone called `set_draw_area()` manually
- Per design the current graphics mode always does full redraws and the reset of the draw area really only adds a few instructions and bytes to be sent over the interface so the performance implications should be negligible compared to sending the full buffer so I'd probably leave it in for simplicity reasons.

Reply:
Ok, thanks for explaining :). Your points make sense. Can you remove the TODO comment and rebase, then I'll get this merged?

Reply:
Done, I've replaced it with a comment explaining why we do it with every `flush()`.